server/leafnode.go

```diff
@@ -499,7 +515,7 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 		// FIXME(dlc) - Make this resolve at startup.
 		acc, err := s.LookupAccount(remote.LocalAccount)
 		if err != nil {
-			c.Debugf("No local account %q for leafnode", remote.LocalAccount)
+			s.Fatalf("No local account %q for leafnode: %v", remote.LocalAccount, err)
```

Comment:
Is this only for local config? Meaning can this fail on operator mode lookup?

Reply:
That's what we discussed yesterday! My initial thought was that we should ERR and retry but you said that the account should always be avail otherwise we should not start (arguably, we could then check the account availability at startup instead of here, but..). 
If you are asking if LookupAccount() can fail, of course it can, if it has to fetch and connection is not avail, etc..

Reply:
We are in createLeafnode() here, which is invoked when soliciting or accepting a LN connection.

Reply:
I think per my comment if we are in local config mode we should lookup any account binding for the leafnode on startup and fail if it does not exist.

Reply:
For solicited ones its essentially on startup, but not for accepted ones.

Reply:
Not true. For solicited one it will happen anytime the connection is broken and server tries to reconnect.
So what should I do? If at startup (in NewServer) we know we are local config mode, then lookup account - ensure it exists - otherwise fail. For other mode, keep lookup here and make ERR and try again?

Reply:
For any leafnode connection there are two opportunities for account binding. We can't control the remote account binding and should not care, I am talking about local account binding mostly for solicited leafnodes with a local configuration for the server.

Reply:
What I mean is that if Server A has a "remote" configuration which specifies a "account" config, then this server will create a LN connection to say Server B. It will do so at startup. But if 10m later B stops and is restarted, Server A will attempt to re-create this solicited connection to B. At that time, it will do a Lookup again...

Reply:
Yes but if Server A has a local config the lookup will either succeed or fail on first incarnation, and if succeeds then will always succeed.

Reply:
My concern was that in operator mode we may not be able to lookup the account at first connect, but may later. But that may be wrong assumption.

Regardless of local/operator mode, lookup on Server A can fail at runtime:
- Account removed and config reload issued
- Account expired
If after any of those 2 events Server B is restarted, Server A would try to recreate connection to B and lookup will fail (just tried with removing account + config reload + restart of B).

Reply:
For Server A that has a solicited connection to Server B, agree that the account can expire, or be removed via config reload. We need to define our desired behavior when that happens. Server should not fail or exit obviously, but probably should not aimlessly try to recreate the connection without a valid local account IMO.

Reply:
All my discussion has been about Server A here.. the one that has a "remote" configuration in "leafnodes{}".
Let's discuss then later, because I am not sure what is the alternative between [FTL] (exit) and [ERR] and try again... except prevent removal/expiration of such account? (not sure how we would do that because that would leave the config file in different state than the runtime)..

Reply:
On startup if a solicited leafnode can not resolve/lookup its local account the server should fail to start IMO. If we subsequently get an error on the account we should shut the leafnode down. We can keep trying to reconnect and failing on account lookup to see if it resolves, or be smarter about it but that feels like we can do that later on.

Reply:
Ok, updated the PR (push forced). Not sure I have captured what you wanted. Happy to iterate if needed or remove the connect/account part and just leave the fix for username/password.